stream/buffered.go

```diff
@@ -0,0 +1,125 @@
+package stream
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/katzenpost/katzenpost/core/worker"
+)
+
+// BufferedStream holds a Stream and Buffer of bytes that are used with cbor.Decoder
+type BufferedStream struct {
```

Comment:
where is this used?

Reply:
BufferedStream is used when using a cbor.Decoder with a Stream: if a CBOR object is incompletely decoded before a Stream is Halt()'d then the contents read from the stream must be preserved in order to resume deserializing CBOR from the Stream in the future. Katzen uses this here:
https://github.com/katzenpost/katzen/blob/3451457574303f0604cc00ec7a0b722653e38c18/katzen.go#L90

Reply:
https://github.com/katzenpost/katzen/blob/3451457574303f0604cc00ec7a0b722653e38c18/katzen.go#L212